eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/SystemObjectModelProvider.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.TypeSpec.Generator.Input;
+using Microsoft.TypeSpec.Generator.Providers;
+using System;
+
+namespace Azure.Generator.Management.Providers
+{
+    internal class SystemObjectModelProvider : ModelProvider
```

Comment:
I think we kind of should not have this type at all, but in our current framework, we have to do this.
Could we create an issue for this or leave some comment here?

Reply:
https://github.com/microsoft/typespec/issues/7162 is tracked for this.